src/Rules/Properties/ReadOnlyByPhpDocPropertyRule.php

```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Properties;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Node\ClassPropertyNode;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+
+/**
+ * @implements Rule<ClassPropertyNode>
+ */
+class ReadOnlyByPhpDocPropertyRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return ClassPropertyNode::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$node->isReadOnlyByPhpDoc()) {
+			return [];
+		}
+
+		$errors = [];
+		if ($node->getDefault() !== null) {
+			 $errors[] = RuleErrorBuilder::message('@readonly property cannot have a default value.')->nonIgnorable()->build();
```

Comment:
maybe this error should be ignorable, not sure. The nonIgnorable is just taken over from `ReadOnlyDocPropertyRule` which deals with native readonly behaviour where this would result in a fatal error.

Reply:
Oh yes, should be ignorable :)